app/controllers/restaurants_controller.rb

```diff
@@ -1,11 +1,15 @@
 class RestaurantsController < ApplicationController
-
   def index
     @restaurants = Restaurant.all
   end
-  
+
   def show
     @restaurant = Restaurant.find(params[:id])
   end
 
+  def rest_cat
```

Comment:
This should really just be handled in the the index action; you can have a conditional so that it looks for the specific category if there is a parameter for category id, otherwise it shows all. Then you can avoid having the extra rest_cat method, and you won't even need an extra route for it in your route files. (Let me know if you don't understand what I'm suggesting and I'll walk you through it.)